User-facing AssertionError with groupby on list of empty list.

I hesitate to post this, but grouping on a list of empty list yields an AssertionError. It's so edge case I'm not sure it's worth bring up though:

``` python
df = pd.DataFrame(zip(range(10), range(10)), columns=['apple', 'b'])
df.groupby([[]])
# Long traceback yields: AssertionError: Grouper and axis must be same length
```

Came from this SO issue (but it's really only related because of the error message) - http://stackoverflow.com/questions/19483991/how-to-make-grouper-and-axis-the-same-length

--- a/pandas/tests/groupby/test_groupby.py
+++ b/pandas/tests/groupby/test_groupby.py
@@ -1714,3 +1714,12 @@ def test_groupby_multiindex_nat():
     result = ser.groupby(level=1).mean()
     expected = pd.Series([3., 2.5], index=["a", "b"])
     assert_series_equal(result, expected)
+
+
+def test_groupby_empty_list_raises():
+    # GH 5289
+    values = zip(range(10), range(10))
+    df = DataFrame(values, columns=['apple', 'b'])
+    msg = "Grouper and axis must be same length"
+    with pytest.raises(ValueError, match=msg):
+        df.groupby([[]])

